scheduler/src/managers/node_agent/metrics.rs

```diff
@@ -35,14 +42,27 @@ pub struct NodeAgent {
 
 impl NodeAgent {
     /// Create a new `NodeAgent`.
-    pub fn new() -> Self {
+    pub fn new(id: String, address: &SocketAddr) -> Self {
         NodeAgent {
+            id,
+            address: *address,
             last_heartbeat: Local::now(),
             memory: None,
             cpu: None,
         }
     }
 
+    /// Get the Agent's id.
```

Comment:
Nit (agent is spelled lowercase everywhere else)
```suggestion
    /// Get the agent's id.
```